### PR TITLE
Removed 30 Scapegoat Errors related to ComparingFloatingPointTypes

### PR DESCRIPTION
**Pull request by carloscaldas**
Closes https://github.com/LBNL-UCB-STI/beam/issues/2990

---
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/2994)

### CODE REVIEW

File: src/main/scala/beam/agentsim/agents/ridehail/TNCIterationStats.scala [hunk 1]
```diff
@@ -449,7 +449,7 @@ case class TNCIterationStats(
       updatedRadius = updatedRadius * 2
     }
 
-    if (circleRadiusInMeters != updatedRadius) {
+    if (circleRadiusInMeters.equals(updatedRadius)) {
```
Comment (rajnikantsh) [inline src/main/scala/beam/agentsim/agents/ridehail/TNCIterationStats.scala:452]: condition is reverse

**Review by rajnikantsh — CHANGES_REQUESTED** [review]
Except 1 condition all LGTM

File: src/main/scala/beam/agentsim/agents/ridehail/TNCIterationStats.scala (hunk 1 above)
Comment (carloscaldas) [reply src/main/scala/beam/agentsim/agents/ridehail/TNCIterationStats.scala]: fixed

**Comment by carloscaldas** [comment]
Test now!

**Comment by carloscaldas** [comment]
Test!

File: src/main/scala/beam/agentsim/agents/choice/mode/RideHailDefaults.scala [hunk 2]
```diff
@@ -10,18 +10,18 @@ import beam.router.model.EmbodiedBeamTrip
   */
 object RideHailDefaults {
   val DEFAULT_COST_PER_MILE = 2.00
-  val zero: Double = 0
+  private val zero: Double = 0D
```
Comment (aaltergot) [inline src/main/scala/beam/agentsim/agents/choice/mode/RideHailDefaults.scala:13]: nit: I would suggest `0.0` format for Double literals (https://www.scala-lang.org/files/archive/spec/2.13/01-lexical-syntax.html).
Also DEFAULT_COST_PER_MILE (line above) should have type annotation as long as it is public.

**Review by aaltergot — APPROVED** [review]

**Review by JustinPihony — APPROVED** [review]

**Review by rajnikantsh — APPROVED** [review]

**Pull request merged**